lib/task.js

```diff
@@ -273,9 +275,43 @@ export class Task {
         .concat(oldEntrypoint);
     }
 
+    if(this.task.payload.features && this.task.payload.features.allowPtrace) {
```

Comment:
And you're absolutely right, the states as they currently stand do not make a lot of sense for this.  The "feature" handling is a very overloaded thing and needs some refactoring.